app/flux/studentStore.js

```diff
@@ -164,6 +164,22 @@ let studentStore = Reflux.createStore({
 		this._loadData(fleshedStudent.id)
 	},
 
+	importStudent(result) {
+		try {
+			let json = undefined
+			json = JSON.parse(result)
+
+			this._preChange()
+			let fleshedStudent = new Student(json)
+			fleshedStudent.save()
+			this._loadData(fleshedStudent.id)
+		}
+		catch (err) {
+			console.error('Error parsing as JSON', result)
+			return
+		}
+	},
```

Comment:
Instead of defining the entire student thing inside the `try`, how about defining `student` before the try, then trying to parse it, then calling the last four things in an if-block predicated on student being defined?